Replace console logger by ILogger when a background service had a critical error and is shutting down the application.

kevgeni opened the pull request:
There was a log of coding style errors in the repo and it would not build. My modification was pretty small so I did not modify those. I suppose there is a refactoring in progress in a different branch.

GeeWee commented:
After having time to look at the PR - the actual changes you're making does break some stuff regarding the `PeriodicTasks` (as they use the `CriticalBackgroundService` internally. We'll need to get that fixed first, luckily it's not too much work.

I don't seem to have edit access to the PR, but I've made some changes and here's two patch files you can apply to your PR that should get it into a working state.

[patches.zip](https://github.com/GeeWee/BetterHostedServices/files/7808983/patches.zip)

kevgeni commented:
Hello there
I applied your patch and merged your latest commits. 
Last time I did this modification without any compilation because it failed for a lot of styling rules that were not my job to fix. I thought you had those errors instead of warnings configured in your project. I was wrong ... it appears that my local .editorconfig was interfering and that's why it did not compile. 
I've moved the project in a different folder and I could see that I produced a broken commit with compilation error :( sry for that ! 

I've added you to my git repo (not sure if I did right, github is not my preferred platform) in case you need to do other modifications like this before merging. If we could have cleared the confusion about why there was some coding errors, I would have probably also fixed it. Hope you like it.
